### PR TITLE
`tail`: obsolete syntax

### PR DESCRIPTION
**Pull request by tertsdiepraam**
Closes https://github.com/uutils/coreutils/issues/2507.

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
MinRustV fails because of `strip_prefix`. This PR should wait until the version bump in https://github.com/uutils/coreutils/pull/2438 is merged, so that issue is resolved.

File: src/uu/tail/src/tail.rs [hunk 1]
```diff
@@ -177,6 +186,60 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
+        if !arg.starts_with(|c: char| c.is_digit(10)) {
+            return false;
+        }
+        let idx = arg.find(|c: char| !c.is_digit(10));
+        let num_slice = match idx {
+            Some(i) => &arg[..i],
+            None => arg,
+        };
+        // Unwrap is safe because we selected only the digits.
+        let num = num_slice.parse::<usize>().unwrap();
```
Comment (miDeb) [inline src/uu/tail/src/tail.rs:204]: Well, _technically_ this could fail if the number is too big.

File: src/uu/tail/src/tail.rs [hunk 2]
```diff
@@ -177,6 +186,60 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
+                    'b' => {
+                        settings.mode = FilterMode::Bytes(num * 512);
+                    }
+                    'c' => {
+                        settings.mode = FilterMode::Bytes(num);
+                    }
+                    'l' => {
+                        settings.mode = FilterMode::Lines(num, b'\n');
+                    }
+                    _ => {}
```
Comment (miDeb) [inline src/uu/tail/src/tail.rs:220]: We also have to return `false` if it is an invalid option.

**Review by miDeb — CHANGES_REQUESTED** [review]
This is not fully correct:
> This obsolete form uses exactly one option and at most one file.

So we have to not treat it as the obsolete setting when there's more than one option or more than one file.

But, wow, all of these rules are so complicated and confusing :sweat_smile:

**Comment by tertsdiepraam** [comment]
@miDeb Thanks for your comments! I'll be able to continue working on this in about a week.

**Comment by miDeb** [comment]
I was trying to implement the obsolete syntax for `sort`, however that doesn't seem to be possible with clap. I was thinking whether it would make sense to not support the obsolete syntax in all of our utilities, for consistency? Maybe supporting the obsolete syntax in some utilities only could be confusing?

**Comment by sylvestre** [comment]
@tertsdiepraam any update on this ? thanks :)

**Comment by tertsdiepraam** [comment]
Ah forgot about it😅 I'll fix this up in a few days.

**Comment by sylvestre** [comment]
no worries ;)

**Comment by sylvestre** [comment]
Did you see this warning?
```

error[E0658]: use of unstable library feature 'bool_to_option'
Error:    --> src/uu/tail/src/tail.rs:264:20
    |
264 |     char.is_none().then(|| ())
    |                    ^^^^
    |
    = note: see issue #64260 <https://github.com/rust-lang/rust/issues/64260> for more information
```

**Comment by tertsdiepraam** [comment]
I've been very busy the past few weeks, so couldn't work on this. I think it's right to close this for now as it presented a lot more challenges than I initially thought.

**Pull request closed without merging**